sidecar/internal/csiaddonsnode/csiaddonsnode.go

```diff
@@ -21,31 +21,63 @@ import (
 	"fmt"
 	"os"
 	"strings"
+	"time"
 
 	csiaddonsv1alpha1 "github.com/csi-addons/kubernetes-csi-addons/api/v1alpha1"
 
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime/serializer"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
 )
 
 const (
 	podNameEnvKey      = "POD_NAME"
 	podNamespaceEnvKey = "POD_NAMESPACE"
 	podUIDEnvKey       = "POD_UID"
 	podIPEnvKey        = "POD_IP"
+
+	// nodeCreationRetry is the delay for calling newCSIAddonsNode after a
+	// failure.
+	nodeCreationRetry = time.Minute * 5
 )
 
 // Deploy creates CSIAddonsNode custom resource with all required information.
+// When information to create the CSIAddonsNode is missing, an error will be
+// returned immediately. If creating the CSIAddonsNode in the Kubernetes
+// cluster fails (missing CRD, object already exists, ...), an error will be
```

Comment:
if `object already exists,` , then we should just continue without error or retries ?

Reply:
If the object exists, it may not be correct (different IP address?), so the container should probably be blocked from starting completely until that is resolved.

Maybe there needs to be a check, that in case the object exists, it is updated instead of created? But that can be a different PR.

Reply:
No, ip address will be unique per pod and will not change during pod lifetime.

If object already exists, then I don't think any parameter will have changed so we just ignore this error and continue.

And in this case, I don't think retrying will help since object will continue to exist until pod is deleted or someone manually deletes the object.

Reply:
Ah, there is a check for that already. I'll clean that up and improve the comment.